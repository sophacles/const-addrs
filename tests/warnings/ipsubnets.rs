use const_addrs::ipsubnets;

fn main() {
    let _ = ipsubnets!("2001:db8::32:ffff", "2001:db8::32:23", 64);
    let _ = ipsubnets!("10.128.0.0", "10.0.0.0", 24);

    //failure so the output is compared
    let _ = ipsubnets!("10128.0.0", "10.0.0.0", 24);
}
