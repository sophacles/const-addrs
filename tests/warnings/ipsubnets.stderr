warning: This range will yeild no values: IP1 >= IP2
 --> tests/warnings/ipsubnets.rs:4:24
  |
4 |     let _ = ipsubnets!("2001:db8::32:ffff", "2001:db8::32:23", 64);
  |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: This iterator requires the lower IP value first in it's arguments list.

warning: This range will yeild no values: IP1 >= IP2
 --> tests/warnings/ipsubnets.rs:5:24
  |
5 |     let _ = ipsubnets!("10.128.0.0", "10.0.0.0", 24);
  |                        ^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: This iterator requires the lower IP value first in it's arguments list.

error: invalid IP address syntax
 --> tests/warnings/ipsubnets.rs:8:24
  |
8 |     let _ = ipsubnets!("10128.0.0", "10.0.0.0", 24);
  |                        ^^^^^^^^^^^
